src/Google/Cache/File.php

```diff
@@ -151,11 +151,14 @@ private function getCacheDir($file, $forWrite)
     // use the first 2 characters of the hash as a directory prefix
     // this should prevent slowdowns due to huge directory listings
     // and thus give some basic amount of scalability
-    $dirHash = substr(md5($file), 0, 2);
+    $fileHash = substr(md5($file), 0, 2);
+    $userHash = md5(get_current_user());
+    $dirHash = $fileHash . DIRECTORY_SEPARATOR . $userHash;
```

Comment:
That doesn't address the problem of the base directory's permissions.  You could solve this by also changing:

```php
if (!mkdir($storageDir, 0700, true)) {
```

to

```php
if (!(mkdir($this->path, 0777, true) && mkdir($storageDir, 0700, true))) {
```